gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1423,7 +1424,16 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
 							checkOrConvertType(store, last, n, &n.Args[0], ct, false)
 						}
 					default:
-						// do nothing
+						ctBase := BaseOf(ct)
+						atBase := baseOf(at)
+
+						_, isCTInterface := ctBase.(*InterfaceType)
+						_, isCTNative := ctBase.(*NativeType)
+						_, isATInterface := atBase.(*InterfaceType)
+
+						if (!isCTInterface && !isCTNative) && isATInterface {
+							panic(fmt.Sprintf("cannot convert %v to %v: need type assertion", at.TypeID(), ct.TypeID()))
+						}
```

Comment:
and it would be better the whole new added logic happens before the `switch`.

Reply:
But then all of these checks will happen needlessly every time.

Reply:
ok to me, thanks.